pyrolite/plot/templates/sandstones.py

```diff
@@ -0,0 +1,134 @@
+import matplotlib.pyplot as plt
+import numpy as np
+
+from ...util.classification import Pettijohn as PJclassifier
+from ...util.classification import Herron as Herronclassifier
+from ...util.log import Handle
+from ...util.meta import sphinx_doi_link, subkwargs, update_docstring_references
+from ...util.plot.axes import init_axes
+
+logger = Handle(__name__)
+
+
+@update_docstring_references
+def Pettijohn(
+    ax=None,
+    add_labels=False,
+    which_labels="ID",
+    relim=True,
+    color="k",
+    **kwargs
+):
+    """
+    Adds the Pettijohn (1973) [#ref_1] sandstones classification diagram.
+
+    Parameters
+    ----------
+    ax : :class:`matplotlib.axes.Axes`
+        Axes to add the template on to.
+    add_labels : :class:`bool`
+        Whether to add labels at polygon centroids.
+    which_labels : :class:`str`
+        Which data to use for field labels - field 'name' or 'ID'.
+    relim : :class:`bool`
+        Whether to relimit axes to fit the built in ranges for this diagram.
+    color : :class:`str`
+        Line color for the diagram.
+
+    Returns
+    -------
+    ax : :class:`matplotlib.axes.Axes`
+
+    References
+    -----------
+    .. [#ref_1] Pettijohn, F. J., Potter, P. E. and Siever, R. (1973).
+                Sand  and Sandstone. New York, Springer-Verlag. 618p.
+                doi: {Pettijohn1973}
+
+    """
+    PJ_xlim, PJ_ylim = (0, 2.5), (-1.5, 1)
+    if ax is None:
+        xlim, ylim = PJ_xlim, PJ_ylim
+    else:
+        # if the axes limits are not defaults, update to reflect the axes
+        ax_defaults = (0, 1)
+        ax_xlim, ax_ylim = ax.get_xlim(), ax.get_ylim()
+        xlim, ylim = (
+            [ax_xlim, PJ_xlim][np.allclose(ax_xlim, ax_defaults)],
+            [ax_ylim, PJ_ylim][np.allclose(ax_ylim, ax_defaults)],
+        )
+    ax = init_axes(ax=ax, **kwargs)
+
+    tas = PJclassifier()
```

Comment:
The variable name here is a bit of a hangover from the TAS class, could consider renaming to `pjc` or similar?